tests/test_config_entries.py

```diff
@@ -1577,3 +1577,198 @@ async def async_step_import(self, user_input):
         assert len(entries) == 1
         assert entries[0].state == config_entries.ENTRY_STATE_LOADED
         assert entries[0].data == {"value": "updated"}
+
+
+async def test_flow_with_default_discovery(hass, manager):
+    """Test that a flow without discovery implementation uses default discovery."""
+    mock_integration(hass, MockModule("comp"))
+    mock_entity_platform(hass, "config_flow.comp", None)
+
+    class TestFlow(config_entries.ConfigFlow):
+        """Test flow."""
+
+        VERSION = 1
+
+        async def async_step_user(self, user_input=None):
+            """Test user step."""
+            return self.async_show_form(step_id="mock")
+
+    with patch.dict(config_entries.HANDLERS, {"comp": TestFlow}):
+        result = await manager.flow.async_init(
+            "comp", context={"source": config_entries.SOURCE_ZEROCONF}
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    flows = hass.config_entries.flow.async_progress()
+    assert len(flows) == 1
+    assert (
+        flows[0]["context"]["unique_id"] == config_entries.DEFAULT_DISCOVERY_UNIQUE_ID
+    )
+
+
+async def test_flow_with_default_discovery_with_unique_id(hass, manager):
+    """Test discovery flow using the default discovery is ignored when unique ID is set."""
+    mock_integration(hass, MockModule("comp"))
+    mock_entity_platform(hass, "config_flow.comp", None)
+
+    class TestFlow(config_entries.ConfigFlow):
+        """Test flow."""
+
+        VERSION = 1
+
+        async def async_step_discovery(self, user_input=None):
+            """Test discovery step."""
+            await self.async_set_unique_id("mock-unique-id")
+            # This call should make no difference, as a unique ID is set
+            await self.async_handle_discovery_without_unique_id()
```

Comment:
This is not a supported use case is it? This is just an internal function?

Reply:
It serves 2 use-cases. I think I've not made that clear enough in my PR description. Hence I've added 2 additional examples (Tellduslive and IPP) to the PR description and implemented that in this PR as well.